### PR TITLE
Problem:(CRO-468) autosync don't work correctly when status reply is not delivered

### PR DESCRIPTION
**Pull request by leejw51crypto**
Solution: in polling , check whether synced or not.
if synced, change the state to waiting
add unit test

apply conditional compilation

### CODE REVIEW

**Comment by leejw51crypto** [comment]
added unit test 
used conditional compilation for simplicity

**Comment by tomtau** [comment]
bors try

File: client-core/src/synchronizer/auto_sync_core.rs [hunk 1]
```diff
@@ -682,4 +707,47 @@ mod tests {
             _ => assert!(false),
         }
     }
+
+    #[test]
+    fn check_chaning_wallets() {
```
Comment (calvinaco) [inline client-core/src/synchronizer/auto_sync_core.rs]:
```suggestion
    fn check_changing_wallets() {
```

Comment (leejw51crypto) [reply client-core/src/synchronizer/auto_sync_core.rs]: thanks!

**Comment by tomtau** [comment]
bors try

**Comment by leejw51crypto** [comment]
rebased

**Comment by tomtau** [comment]
> rebased

@leejw51crypto and squash the commits: https://github.com/wprig/wprig/wiki/How-to-squash-commits ?

**Comment by devashishdxt** [comment]
@tomtau: This PR also contains HD wallet related changes. Is there any way to revert this merge?

**Comment by tomtau** [comment]
@devashishdxt yeah, I noticed later that this contains some other random changes

**Comment by tomtau** [comment]
reverted in https://github.com/crypto-com/chain/commit/6bc3901325ecc0197fc245c028d86d60d12b873a

@leejw51 please open the PR again with the auto sync + test changes ONLY

**Comment by tomtau** [comment]
this PR unfortunately contained random HD wallet changes + old master

**Pull request merged**